### ISSUE TITLE
Fix `declaration-property-value-no-unknown` false positives for descriptors

### ISSUE THREAD

**Issue by jeddy3** [stylelint/stylelint#8150]
### What minimal example or steps are needed to reproduce the bug?

```css
@property --foo {
  syntax: bar;
}
```

### What minimal configuration is needed to reproduce the bug?

```json
{
  "rules": {
    "declaration-property-value-no-unknown": true
  }
}
```

### How did you run Stylelint?

[Demo](https://stylelint.io/demo/#N4Igxg9gJgpiBcIACAHAThFM0BcCeABALREBmEEBwAOgHYEEDOetOAhgB7wEBGbaAbjoBfEABoQpAJYAbGADk2AWziIYHZSjkA6MI0bjwEWtIDmCEDXoFqINAFc5jW9ysMGt2GBn82OKcZE6JjY+EQAbmwy9jBEtBBE9rQA1vEA7rQuBDgOMHQMwiKGkCZSpgBiEGhKfhYAVozGhrAoBohuNiCM+HIyUqxZtj44MN22Yvmd3Xi9-ThEJWZE3Wy0UPxQgyDDozi2RcJAA)

### Which Stylelint-related dependencies are you using?

[stylelint](https://stylelint.io/)@16.11.0
[stylelint-config-standard](https://www.npmjs.com/package/stylelint-config-standard)@36.0.1

### What did you expect to happen?

No problems, as it's a descriptor, not a property.

### What actually happened?

The following:

```console
2:11-14 error Unexpected unknown value "bar" for property "syntax" (declaration-property-value-no-unknown)
```

### Do you have a proposal to fix the bug?

Only check properties in the rule.

**Comment by ybiquitous** [comment]
According to the [MDN document](https://developer.mozilla.org/en-US/docs/Web/CSS/@property/syntax), the `syntax` descriptor accepts only a string value like `syntax: "<color>"`. Is a case like `syntax: bar` valid?

**Comment by jeddy3** [comment]
This fix narrows the scope of the rule so that it doesn't overlap with the new `at-rule-descriptor-value-no-unknown` one (https://github.com/stylelint/stylelint/issues/8149).

Currently, the `declaration-property-value-no-unknown` rule checks properties and descriptors. This is potentially confusing for users for things like `font-weight`, which is the name of a [descriptor](https://csstree.github.io/docs/syntax/#AtruleDescriptor:font-face%2Ffont-weight) and a [property](https://csstree.github.io/docs/syntax/#Property:font-weight) that accept different things.

For example:

```css
@font-face {
  font-weight: bolder;
}
```

Currently gives:

```console
2:11-14 error Unexpected unknown value "bolder" for property "font-weight" (declaration-property-value-no-unknown)
```

`bolder` is a known value for the `font-weight` _property_ but not the `font-weight` _descriptor_.

It'd be better if this was only caught by the new `at-rule-descriptor-value-no-unknown` rule:

```console
2:11-14 error Unexpected unknown value "bolder" for descriptor "font-weight" (at-rule-descriptor-value-no-unknown)
```

**Comment by ybiquitous** [comment]
I now understand the difference between properties and descriptors. I agree with this suggested fix. 👍🏼 

In addition, should we clarify this difference on the `declaration-property-value-no-unknown` rule document? Or is it unnecessary since the difference may be obvious?

**Comment by jeddy3** [comment]
> Should we clarify this difference on the declaration-property-value-no-unknown rule document?

I think it's a good idea. Something along the lines of "This rule does not check the value of at-rule descriptors. You can use the `at-rule-descriptor-value-no-unknown` to check those."
